client-spark/spark-3/src/main/scala/org/apache/spark/shuffle/celeborn/CelebornShuffleReader.scala

```diff
@@ -214,46 +219,71 @@ class CelebornShuffleReader[K, C](
             locations,
             streamHandlers,
             fileGroups.mapAttempts,
-            metricsCallback)
+            metricsCallback,
+            chunkPrefetchEnabled)
+          streams.put(partitionId, inputStream)
         } catch {
           case e: IOException =>
             logError(s"Exception caught when readPartition $partitionId!", e)
             exceptionRef.compareAndSet(null, e)
-            null
           case e: Throwable =>
             logError(s"Non IOException caught when readPartition $partitionId!", e)
             exceptionRef.compareAndSet(null, new CelebornIOException(e))
-            null
         }
-      } else null
+      }
     }
 
+    (startPartition until Math.min(
```

Comment:
Why we need a creation window for this?

Reply:
That's because when chunk prefetch is enabled, inputstream will create chunk buffer in construction, the purpose of window size is to control number of inputstreams to avoid OOM.

Reply:
It's related to https://github.com/apache/celeborn/pull/2348

Reply:
Adding a limit that a task should prefetch to control the memory usage can be better.

16 may not be suitable for all scenarios. And this config will be hacky to tweak.